### PR TITLE
Clean tests for repositories using Mockito

### PR DESCRIPTION
**Pull request by Ph0tonic**

### CODE REVIEW

File: app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt [hunk 1]
```diff
@@ -5,16 +5,25 @@ import androidx.lifecycle.MutableLiveData
...
+        TODO("Not yet implemented")
+    }
+
+    override fun removeAllUserOfSearchGroup(searchGroupId: String) {
+        TODO("Not yet implemented")
     }
 
+    override fun addUserToSearchGroup(searchGroupId: String, user: UserData) {
+        TODO("Not yet implemented")
+    }
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt:24]: Are those going to get implemented or is  an oversight ?

File: app/src/androidTest/java/ch/epfl/sdp/database/repository/HeatmapRepositoryTest.kt [hunk 2]
```diff
@@ -22,88 +20,45 @@ class HeatmapRepositoryTest {
     val instantTaskExecutorRule = InstantTaskExecutorRule()
 
     companion object {
-        private val DUMMY_LOCATION_1 = LatLng(13.0, 42.0)
-        val DUMMY_LOCATION_2 = LatLng(12.0, 42.0)
-        private const val DUMMY_INTENSITY_1 = 666.0
-        private const val DUMMY_INTENSITY_2 = 333.0
         private const val DUMMY_HEATMAP_ID = "dummy heatmap id"
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/database/repository/HeatmapRepositoryTest.kt:23]: Maybe put underscores here too for consistency

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 3]
```diff
@@ -70,19 +85,28 @@ class DroneStatusFragmentTest {
...
         val targetContext: Context = InstrumentationRegistry.getInstrumentation().targetContext
         preferencesEditor = PreferenceManager.getDefaultSharedPreferences(targetContext).edit()
     }
 
-
     @Test
     fun updateDroneBatteryChangesDroneStatus() {
         mActivityRule.launchActivity(intentWithGroupAndOperator)
+        mUiDevice.wait(Until.hasObject(By.desc(mActivityRule.activity.getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
+        assertThat(mActivityRule.activity.mapView.contentDescription == mActivityRule.activity.getString(R.string.map_ready), equalTo(true))
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:109]: Do we really need to wait thath the map displays here ? The status bar should show and update even without the map, or am I wrong ?

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 4]
```diff
@@ -108,6 +132,8 @@ class DroneStatusFragmentTest {
     @Test
     fun updateDroneAltitudeChangesDroneStatus() {
         mActivityRule.launchActivity(intentWithGroupAndOperator)
+        mUiDevice.wait(Until.hasObject(By.desc(mActivityRule.activity.getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
+        assertThat(mActivityRule.activity.mapView.contentDescription == mActivityRule.activity.getString(R.string.map_ready), equalTo(true))
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:136]: Sqme question here

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 5]
```diff
@@ -134,6 +160,8 @@ class DroneStatusFragmentTest {
     @Test
     fun updateDroneSpeedChangesDroneStatus() {
         mActivityRule.launchActivity(intentWithGroupAndOperator)
+        mUiDevice.wait(Until.hasObject(By.desc(mActivityRule.activity.getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
+        assertThat(mActivityRule.activity.mapView.contentDescription == mActivityRule.activity.getString(R.string.map_ready), equalTo(true))
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:164]: Sqme question here

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 6]
```diff
@@ -160,6 +188,9 @@ class DroneStatusFragmentTest {
     @Test
     fun updateDronePositionChangesDistToUser() {
         mActivityRule.launchActivity(intentWithGroupAndOperator)
+        mUiDevice.wait(Until.hasObject(By.desc(mActivityRule.activity.getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
+        assertThat(mActivityRule.activity.mapView.contentDescription == mActivityRule.activity.getString(R.string.map_ready), equalTo(true))
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:192]: Sqme question here

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 7]
```diff
@@ -175,6 +206,9 @@ class DroneStatusFragmentTest {
     @Test
     fun updateUserPositionChangesDistToUser() {
         mActivityRule.launchActivity(intentWithGroupAndOperator)
+        mUiDevice.wait(Until.hasObject(By.desc(MainApplication.applicationContext().getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
+        assertThat(mActivityRule.activity.mapView.contentDescription == MainApplication.applicationContext().getString(R.string.map_ready), equalTo(true))
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:210]: Sqme question here

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 8]
```diff
@@ -190,6 +224,9 @@ class DroneStatusFragmentTest {
     @Test
     fun updateBatteryLevelChangesBatteryLevelIcon() {
         mActivityRule.launchActivity(intentWithGroupAndOperator)
+        mUiDevice.wait(Until.hasObject(By.desc(mActivityRule.activity.getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
+        assertThat(mActivityRule.activity.mapView.contentDescription == mActivityRule.activity.getString(R.string.map_ready), equalTo(true))
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:228]: Sqme question here

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 9]
```diff
@@ -229,6 +266,9 @@ class DroneStatusFragmentTest {
     @Test
     fun updateDronePositionChangesDistToHome() {
         mActivityRule.launchActivity(intentWithGroupAndOperator)
+        mUiDevice.wait(Until.hasObject(By.desc(mActivityRule.activity.getString(R.string.map_ready))), MAP_LOADING_TIMEOUT)
+        assertThat(mActivityRule.activity.mapView.contentDescription == mActivityRule.activity.getString(R.string.map_ready), equalTo(true))
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:270]: Sqme question here

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt [hunk 10]
```diff
@@ -244,18 +284,30 @@ class DroneStatusFragmentTest {
...
         onView(withId(R.id.distance_to_home)).check(matches(withText(DEFAULT_ALTITUDE_DISPLAY)))
+        runOnUiThread { Timber.w("Step 7") }
 
         runOnUiThread {
             Drone.homeLocationLiveData.value = Telemetry.Position(1.0, 0.0, 0f, 0f)
+            runOnUiThread { Timber.w("Step 8") }
         }
 
         onView(withId(R.id.distance_to_home)).check(matches(not(withText(DEFAULT_ALTITUDE_DISPLAY))))
+        runOnUiThread { Timber.w("Step 9") }
```
Comment (Dragoo417) [inline app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:311]: Sqme question here. Also I think it would be good to remove the logs

Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:311]: Yes it was for debug purposes and I forgot to remove them.

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt (hunk 9 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:270]: Correct, it was a attempt to solve an issue. I'll remove it.

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt (hunk 3 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt:109]: Correct, it was a attempt to solve an issue. I'll remove it.

File: app/src/androidTest/java/ch/epfl/sdp/database/repository/HeatmapRepositoryTest.kt (hunk 2 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/database/repository/HeatmapRepositoryTest.kt:23]: I'll fix it.

File: app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt (hunk 1 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt:24]: They can be implemented later, they are not necessary yet.

File: app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt (hunk 8 above)
Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/ui/drone/DroneStatusFragmentTest.kt]: Correct, it was a attempt to solve an issue. I'll remove it.

File: app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt [hunk 11]
```diff
@@ -5,16 +5,25 @@ import androidx.lifecycle.MutableLiveData
 import ch.epfl.sdp.database.data.Role
 import ch.epfl.sdp.database.data.UserData
 
-open class EmptyMockUserDao : UserDao {
+class MockUserDao : UserDao {
```
Comment (Sporarum) [inline app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt:8]: Couldn't this file also be replaced by a mockito mock ?

Comment (Ph0tonic) [reply app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt:8]: This files aims to be a real offline implementation of the dao for test purposes.

Comment (Sporarum) [reply app/src/androidTest/java/ch/epfl/sdp/database/dao/MockUserDao.kt:8]: Okay, thank you for the explanation

**Pull request merged**